mmcv/ops/csrc/pytorch/cuda/bias_act_cuda.cu

```diff
@@ -0,0 +1,282 @@
+      }
+      if (G == 1) y = x * (one - yy * yy);
+      if (G == 2) y = x * (one - yy * yy) * (-two * yy);
+    }
+
+    // sigmoid
+    if (A == 5) {
+      if (G == 0) y = (x < -expRange) ? 0 : one / (exp(-x) + one);
+      if (G == 1) y = x * yy * (one - yy);
+      if (G == 2) y = x * yy * (one - yy) * (one - two * yy);
+    }
+
+    // elu
+    if (A == 6) {
+      if (G == 0) y = (x >= 0) ? x : exp(x) - one;
+      if (G == 1) y = (yy >= 0) ? x : x * (yy + one);
+      if (G == 2) y = (yy >= 0) ? 0 : x * (yy + one);
+    }
+
+    // selu
+    if (A == 7) {
+      if (G == 0)
+        y = (x >= 0) ? seluScale * x : (seluScale * seluAlpha) * (exp(x) - one);
+      if (G == 1)
+        y = (yy >= 0) ? x * seluScale : x * (yy + seluScale * seluAlpha);
+      if (G == 2) y = (yy >= 0) ? 0 : x * (yy + seluScale * seluAlpha);
+    }
+
+    // softplus
+    if (A == 8) {
+      if (G == 0) y = (x > expRange) ? x : log(exp(x) + one);
+      if (G == 1) y = x * (one - exp(-yy));
+      if (G == 2) {
+        scalar_t c = exp(-yy);
+        y = x * c * (one - c);
+      }
+    }
+
+    // swish
+    if (A == 9) {
+      if (G == 0)
+        y = (x < -expRange) ? 0 : x / (exp(-x) + one);
+      else {
+        scalar_t c = exp(xref);
+        scalar_t d = c + one;
+        if (G == 1)
+          y = (xref > halfExpRange) ? x : x * c * (xref + d) / (d * d);
+        else
+          y = (xref > halfExpRange)
+                  ? 0
+                  : x * c * (xref * (two - d) + two * d) / (d * d * d);
+        yref = (xref < -expRange) ? 0 : xref / (exp(-xref) + one) * gain;
+      }
+    }
+
+    // Apply gain.
+    y *= gain * dy;
+
+    // Clamp.
+    if (clamp >= 0) {
+      if (G == 0)
+        y = (y > -clamp & y < clamp) ? y : (y >= 0) ? clamp : -clamp;
+      else
+        y = (yref > -clamp & yref < clamp) ? y : 0;
+    }
+
+    // Store.
+    ((T*)p.y)[xi] = (T)y;
+  }
+}
+
+//------------------------------------------------------------------------
+// CUDA kernel selection.
+
+template <class T>
+void* choose_bias_act_kernel(const bias_act_kernel_params& p) {
+  if (p.act == 1) return (void*)bias_act_kernel<T, 1>;
+  if (p.act == 2) return (void*)bias_act_kernel<T, 2>;
+  if (p.act == 3) return (void*)bias_act_kernel<T, 3>;
+  if (p.act == 4) return (void*)bias_act_kernel<T, 4>;
+  if (p.act == 5) return (void*)bias_act_kernel<T, 5>;
+  if (p.act == 6) return (void*)bias_act_kernel<T, 6>;
+  if (p.act == 7) return (void*)bias_act_kernel<T, 7>;
+  if (p.act == 8) return (void*)bias_act_kernel<T, 8>;
+  if (p.act == 9) return (void*)bias_act_kernel<T, 9>;
+  return NULL;
+}
+
+//------------------------------------------------------------------------
+// Template specializations.
+
+template void* choose_bias_act_kernel<double>(const bias_act_kernel_params& p);
+template void* choose_bias_act_kernel<float>(const bias_act_kernel_params& p);
+template void* choose_bias_act_kernel<c10::Half>(
+    const bias_act_kernel_params& p);
+
```

Comment:
It is Ok to remove these explicit specialization if the template is in included or in the same file.